notebooks/5_ModelComparison/1_Ins_ModelComparison.ipynb

```diff
@@ -80,41 +80,103 @@
+    "\n",
+    "\n",
+    "\\*Note:* This section heavily draws from Richard McElreath's examples. We'll be covering the material at a high level to get a full understanding we highly recommend his  [lectures](https://www.youtube.com/watch?v=gjrsYDJbRh0) and [books](https://www.amazon.com/Statistical-Rethinking-Bayesian-Examples-Chapman/dp/1482253445/). \n",
+    "\n",
+    "\n",
+    "To restate our prior question \"If have  3, or 5, or 10, estimates of the truth (also called models) which most usefully explains the truth, given the truth is also uncertain?\"\n",
+    "\n",
+    "Over the years a number of folks have contributed mathematical tools that help answer this question. In short the we go in this order\n",
+    "\n",
+    "1. How do we quantify uncertainty? (Information Entropy)\n",
+    "2. How do we quantify the additional uncertainty added from an estimate when we know the truth? (KL Divergence)\n",
+    "3. How do we estimate the additional uncertainty added from an estimate when we don't know the truth? (Deviance)\n",
+    "4. How do we make sure we're ensure were underestimating uncertainty with complex models? (Information Criteria)\n",
+    "\n",
+    "This section is quite heavy of math. However it is not critical to understand the fine details, but more so the path of thinking.\n",
+    "\n",
+    "#### *Information Entropy*: Quantifying Uncertainty\n",
+    "Weather is uncertain, and dice rolls are uncertain. But weather in certain areas is delightfully consistent, like my hometown of Orange County, and \"loaded\" are designed to be inconsistent.\n",
+    "\n",
+    "Information Entropy is a summarization of our uncertainty through the following formula  \n",
+    "$$ H(p) = -\\sum_{n=1}^{N} p_{i}log(p_{i})$$\n",
+    "\n",
+    "#### *Divergence*: Quantifying additional uncertainty added from estimations of the truth\n",
+    "Now that we can quantify the uncertainty in the distribution of reality, we can compare the uncertainty in the distribution our model to see how far off we are. The formula is as follows, where $p$ is the truth and $q$ is the estimate\n",
+    "\n",
+    "$$D_{kl} = -\\sum_{n=1}^{N} p_{i}(log(p_{i} - log(q_{i}))$$\n",
     "\n",
-    "* PSIS-Loo\n",
-    "* Widely Applicable Information Criteria\n",
-    "* az.plot_compare"
+    "This is the idea of **divergence**, a measurement of the \"distance\" between two distributions. One important note this distance is not symmetric. For some quick intuition It was way more surprising for me to go from always sunny Los Angeles and experience snow in Wisconsin, than it is for a native Wisconsite to see sun in Los Angeles. People in Los Angeles only see sun, people in Wisconsin see sun and snow, therefore a Los Angeles is much farther removed from the reality of a mixed sunny/snowy climate, than a Wisconsinite is to an always sunny climate.\n",
+    "\n",
+    "#### *Deviance*: Estimating additional uncertainty because we don't know the truth using Deviance\n",
```

Comment:
what about showing eq 5.18 (From Bayesian analysis with Python)?

Reply:
I guess what I still don't understand is that deviance can be calculated with just one model fit, but in 5.18 two distributions (q and r) are needed to obtain a difference.

Richard McElreath puts this formula as an estimation of deviance which only requires one distribution which I still don't have a good intuition for
<img width="733" alt="image" src="https://user-images.githubusercontent.com/7213793/58668965-9e47e680-8308-11e9-932c-4043ce44fb38.png">

Reply:
The trick is that you can get rid of `p` because is a common factor.

The deviance comes from 5.18, assuming we do not care about the common `p`. If you _just delete_ `p` you will get the deviance of `q` minus the deviance of `r`  (also 5.18 is missing the "-2" factor that we can also omit because is a constant). From this we can see that by itself the deviance is not useful, only when you compare two deviances (for the same data) is that you can say something about how good the models are.

Reply:
That much makes sense. What I'm confused about is how we're able to compute IC, which requires deviance, if we just fit one model? With Richard's formula this is possible, but with formula 5.18 I'm confused

How does Richard get away with only one distribution?

<img width="835" alt="image" src="https://user-images.githubusercontent.com/7213793/58671403-85433380-8310-11e9-9411-4ad7a677f3ae.png">